moon-store-mac-os/moon-store-mac-os/Domain/Repositories/AuthenticationRepository.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  LoginRepository.swift
+//  moon-store-mac-os
+//
+//  Created by Jose Luna on 12/14/24.
+//
+
+import Foundation
+
+final class AuthenticationRepository: BaseNetworkService {
+    var loggedUser: UserModel? {
+        guard let storedUser = try? store.fetch().first else { return nil }
+        return UserModel.factory(from: storedUser)
```

Comment:
@Rexmoon When you use a `factory` method to create a new instance, you are increasing the stack function calls unnecessarily, for performace, the best way is to create the instance directly in the line:

```suggestion
        return UserModel(...)
        
        // or
        
        JsonEncoder().encode(UserModel.self, data: storedUser)
```